app/models/grda_warehouse/hud/client.rb

```diff
@@ -1714,6 +1714,31 @@ def last_projects_served_by(include_confidential_names: false)
       end.uniq.sort
     end
 
+    def last_intentional_contacts(user, include_confidential_names: false, skip_confidential_projects: false, include_dates: false)
+      contacts = processed_service_history&.last_intentional_contacts
+      return [] unless contacts.present?
+
+      visible_projects = user.visible_project_ids_enrollment_context
+      contacts = JSON.parse(contacts)
+      projects = GrdaWarehouse::Hud::Project.find(contacts.map { |contact| contact['project_id'] }.compact).index_by(&:id)
+
+      contacts.map do |contact|
+        project_id = contact['project_id']
+        next if project_id.present? && !project_id.in?(visible_projects)
+
+        confidential = projects[project_id]&.confidential? || false
+        next if confidential && skip_confidential_projects
+
+        name = if ! confidential || include_confidential_names
+          contact['project_name']
+        else
+          GrdaWarehouse::Hud::Project.confidential_project_name
+        end
```

Comment:
I think this can call `project.name(...)` with the appropriate parameters.  Let's keep the logic of how to display the name in Project.